### PR TITLE
Fix burning gibs

### PR DESCRIPTION
**Pull request by SnowshoeIceboot**
Source SDK 2013 has code to attach particles to burning gibs, which live TF2 uses. The TF2 source code we use is over 8 years old, so you can imagine why it's not working when you try to turn it on.

I updated C_TFRagdoll::CreateTFGibs to check if the player is burning. The burning state is then passed on to C_TFPlayer::CreatePlayerGibs to check if bBurning parameter in CreateGibsFromList should be enabled.

As long as we are using an up to date burningplayer.pcf with the "burninggibs" particle (which I think we are, live TF2 has it), it should work. You will need cl_burninggibs 1 in the console to see it, as it's not enabled by default.

screen1
screen2

### CODE REVIEW

File: src/game/client/tf/c_tf_player.cpp [hunk 1]
```diff
@@ -549,7 +549,14 @@ void C_TFRagdoll::CreateTFGibs( void )
 	{
 		Vector vecVelocity = m_vecForce + m_vecRagdollVelocity;
 		VectorNormalize( vecVelocity );
-		pPlayer->CreatePlayerGibs( m_vecRagdollOrigin, vecVelocity, m_vecForce.Length() );
+		if ( m_bBurning )
+		{
+			pPlayer->CreatePlayerGibs( m_vecRagdollOrigin, vecVelocity, m_vecForce.Length(), true );
```
Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: You can just do pPlayer->CreatePlayerGibs( m_vecRagdollOrigin, vecVelocity, m_vecForce.Length(), m_bBurning );

File: src/game/client/tf/c_tf_player.h [hunk 2]
```diff
@@ -138,7 +138,7 @@ class C_TFPlayer : public C_BasePlayer, public IHasAttributes
 
 	// Gibs.
 	void InitPlayerGibs( void );
-	void CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVelocity, float flImpactScale );
+	void CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVelocity, float flImpactScale, bool bBurning );
```
Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.h]: You should add a default parm here

File: src/game/client/tf/c_tf_player.cpp [hunk 3]
```diff
@@ -3461,7 +3468,14 @@ void C_TFPlayer::CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVel
 
 	// Break up the player.
 	m_hSpawnedGibs.Purge();
-	m_hFirstGib = CreateGibsFromList( m_aGibs, GetModelIndex(), NULL, breakParams, this, -1 , false, true, &m_hSpawnedGibs );
+	if ( bBurning )
+	{
+		m_hFirstGib = CreateGibsFromList( m_aGibs, GetModelIndex(), NULL, breakParams, this, -1 , false, true, &m_hSpawnedGibs, true );
```
Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: Same here

File: src/game/client/tf/c_tf_player.cpp [hunk 4]
```diff
@@ -3461,7 +3468,14 @@ void C_TFPlayer::CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVel
 
 	// Break up the player.
 	m_hSpawnedGibs.Purge();
-	m_hFirstGib = CreateGibsFromList( m_aGibs, GetModelIndex(), NULL, breakParams, this, -1 , false, true, &m_hSpawnedGibs );
+	if ( bBurning )
```
Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: Not sure why you pass the parameter through instead of just checking m_bBurning here?

Comment (SnowshoeIceboot) [inline src/game/client/tf/c_tf_player.cpp]: It's a private bool in C_TFRagdoll

'm_bBurning' : undeclared identifier

Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: Ok, yeah, thats fine then

File: src/game/client/tf/c_tf_player.h (hunk 2 above)
Comment (SnowshoeIceboot) [inline src/game/client/tf/c_tf_player.h]: Example of what you mean?

Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.h]: void CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVelocity, float flImpactScale, bool bBurning = false);

File: src/game/client/tf/c_tf_player.cpp (hunk 3 above)
Comment (SnowshoeIceboot) [inline src/game/client/tf/c_tf_player.cpp]: Are you talking about CreateGibsFromList?

Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: I am saying, you should just call CreateGibsFromList directly.

Comment (SnowshoeIceboot) [inline src/game/client/tf/c_tf_player.cpp]: I don't understand what you mean.

Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: You don't need the if statement. You can just do m_hFirstGib = CreateGibsFromList( m_aGibs, GetModelIndex(), NULL, breakParams, this, -1 , false, true, &m_hSpawnedGibs, bBurning );

Comment (SnowshoeIceboot) [inline src/game/client/tf/c_tf_player.cpp]: Oh, duh. Sorry. I'll fix this up.

**Comment by SnowshoeIceboot** [comment]
Alright, check #261 3a36289

File: src/game/client/tf/c_tf_player.cpp [hunk 5]
```diff
@@ -3436,7 +3436,7 @@ void C_TFPlayer::InitPlayerGibs( void )
 //			&vecVelocity - 
 //			&vecImpactVelocity - 
 //-----------------------------------------------------------------------------
-void C_TFPlayer::CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVelocity, float flImpactScale )
+void C_TFPlayer::CreatePlayerGibs( const Vector &vecOrigin, const Vector &vecVelocity, float flImpactScale, bool bBurning = false )
```
Comment (PistonMiner) [inline src/game/client/tf/c_tf_player.cpp]: Default parms should be in the header file (only).

**Comment by SnowshoeIceboot** [comment]
Oops, fixed.

**Pull request merged**